internal/btf/btf_test.go

```diff
@@ -34,13 +34,61 @@ func parseVmLinux(tb testing.TB) (*Spec, error) {
 	return spec, nil
 }
 
-func TestFindType(t *testing.T) {
-	spec, err := parseVmLinux(t)
+func parseELFBTF(tb testing.TB, file string) (*Spec, error) {
+	fh, err := os.Open(file)
+	if err != nil {
+		tb.Fatal(err)
+	}
+	defer fh.Close()
+
+	elf, err := internal.NewSafeELFFile(fh)
+	if err != nil {
+		tb.Fatal(err)
+	}
+
+	sec := elf.Section(".BTF")
+	if sec == nil {
+		tb.Fatalf("ELF %s does not contain .BTF section", file)
+	}
+	rd := sec.Open()
+
+	return loadRawSpec(rd, elf.ByteOrder, nil, nil)
+}
+
+func TestAnyTypesByName(t *testing.T) {
+	testutils.Files(t, testutils.Glob(t, "testdata/relocs-*.elf"), func(t *testing.T, file string) {
+		spec, err := parseELFBTF(t, file)
+		if err != nil {
+			t.Fatalf("parsing ELF BTF: %v", err)
+		}
+
+		types, err := spec.AnyTypesByName("ambiguous")
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		if len(types) != 2 {
+			t.Fatalf("expected to receive exactly 2 types from querying ambiguous type, got: %v", types)
+		}
+
+		types, err = spec.AnyTypesByName("ambiguous___flavour")
```

Comment:
Can you add a similar flavour test for `TypeByName`?

Reply:
Done. 64a9d86 This at least encodes the behaviour you've discussed in https://github.com/cilium/ebpf/pull/503#discussion_r764924572, but we can change this later to include a priority system of some sort.